syncTextures.js

```diff
@@ -168,26 +167,34 @@ function verifyTextureFolders(dirList) {
  * @return {Promise} 
  */
 async function downloadFile(url, local, agent) {
+  const writer = fs.createWriteStream(local);
+
   let config = {
     responseType: 'stream',
     httpsAgent: agent,
   };
 
-  await axios.get(url, config)
-    .then(response => {
-      response.data.pipe(fs.createWriteStream(local));
+  return axios.get(url, config)
+  .then(response => {
+    return new Promise((resolve, reject) => {
+      response.data.pipe(writer);
       console.log('Downloading ' + url);
-    })
-    .catch(function (error) {
-      console.log(error.toJSON());
-    })
-    .then(function () {
-      agent.destroy()
+      let error;
+      writer.on('error', err => {
+        error = err;
+        writer.close();
+        reject(err);
+      });
+      writer.on('close', () => {
```

Comment:
separating out the file writer allows us to track when the file is complete (poor code on my part, I should have caught this and re-did this from the beginning)